src/Sentry.AspNetCore/SentryHostBuilderExtensions.cs

```diff
@@ -0,0 +1,107 @@
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(this IHostBuilder builder, string dsn)
+            => builder.UseSentry(o => o.Dsn = dsn);
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureOptions">The configure options.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<SentryAspNetCoreOptions>? configureOptions)
+            => builder.UseSentry((_, options) => configureOptions?.Invoke(options));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureOptions">The configure options.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, SentryAspNetCoreOptions>? configureOptions)
+            => builder.UseSentry((context, sentryBuilder) =>
+                sentryBuilder.AddSentryOptions(options => configureOptions?.Invoke(context, options)));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureSentry">The Sentry builder.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<ISentryBuilder>? configureSentry) =>
+            builder.UseSentry((_, sentryBuilder) => configureSentry?.Invoke(sentryBuilder));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureSentry">The Sentry builder.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, ISentryBuilder>? configureSentry)
+        {
+            // The earliest we can hook the SDK initialization code with the framework
+            // Initialization happens at a later time depending if the default MEL backend is enabled or not.
+            // In case the logging backend was replaced, init happens later, at the StartupFilter
+            _ = builder.ConfigureLogging((context, logging) =>
+            {
+                logging.AddConfiguration();
+
+                var section = context.Configuration.GetSection("Sentry");
+                _ = logging.Services.Configure<SentryAspNetCoreOptions>(section);
+
+                _ = logging.Services
+                    .AddSingleton<IConfigureOptions<SentryAspNetCoreOptions>, SentryAspNetCoreOptionsSetup>();
+                _ = logging.Services.AddSingleton<ILoggerProvider, SentryAspNetCoreLoggerProvider>();
```

Comment:
@bruno-garcia using TryAddSingleton here can cause issues, as it won't register it if let's say Serilog or other already exists, in this case you want the 'override' ability

Reply:
That's true. Is there a way to check if the specific specification has been added? I believe it was possible to check the items in the collection so we can verify if specifically `SentryAspNetCoreLoggerProvider` was already added and make sure we only add it once.